### DISCUSSION TITLE
msgfmt: cannot locate ITS rules

### DISCUSSION THREAD

**Discussion by duzhuqi** [cockpit-project/cockpit#17524] [General]
### Explain what happens

When COMPILING conckpit, I reported the following error, how to solve the problem? The operating system I use is CentOS8 and the kernel version is 4.18.0-394.el8.x86_64.

  GEN      src/ws/cockpit.appdata.xml
  GEN      pkg/sosreport/org.cockpit-project.cockpit-sosreport.metainfo.xml
msgfmt: cannot locate ITS rules for pkg/sosreport/org.cockpit-project.cockpit-sosreport.metainfo.xml.in
make[1]: *** [Makefile:8849：pkg/sosreport/org.cockpit-project.cockpit-sosreport.metainfo.xml] Error 1

### Version of Cockpit

272

### Server operating system

CentOS

### Server operating system version

CentOS8

**Comment by garrett** [comment]
Firstly: [CentOS 8 went out of support in December 2021](https://www.centos.org/centos-linux-eol/). If you're using CentOS 8, you should [switch to CentOS Stream 8](https://www.centos.org/news-and-events/convert-to-stream-8/), which is easy with these two commands (as root):

```sh
dnf --disablerepo '*' --enablerepo extras swap centos-linux-repos centos-stream-repos
dnf distro-sync
```

Once you're on CentOS Stream 8 — or if you already are — then you can install the dependencies for Cockpit. To install `libappstream-glib-devel` (which is the specific package you need, according to your error), it requires EPEL and powerttools. You can then use `builddep` for Cockpit, so all the build dependencies are met.

Here are commands (that you have to run as root) which set up EPEL, then powertools, and then install Cockpit's build dependencies:

```sh
dnf install -y epel-release
dnf config-manager --set-enabled powertools
dnf builddep -y cockpit
```

I hope this helps!
